services/bettersocial/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,22 @@
+const {messaging} = require('firebase-admin');
+const FcmTokenFunction = require('../../../databases/functions/fcmToken');
+const {FcmToken, User} = require('../../../databases/models');
+const UsersFunction = require('../../../databases/functions/users');
+
+const sendChatNotification = async (userTargetId, payload, options = {}) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
+  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(FcmToken, signedUser);
+  if (userTargetToken) {
+    try {
+      userTargetToken.forEach((user) => {
+        messaging()
+          .sendToDevice(user?.token, payload, options)
+          .then(() => {});
+      });
+    } catch (error) {
+      console.log('error send chat notifications', error);
+    }
+  }
+};
+
+module.exports = sendChatNotification;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendChatNotification` function has a few issues that need to be addressed:

1. Error handling: The current error handling strategy is not robust enough. It simply logs the error to the console and does not provide any feedback to the caller of the function. This could lead to silent failures where an error occurs but the calling code is unaware of it.

2. Promise handling: In line 13, you're sending messages to devices using Firebase's `sendToDevice` method which returns a promise. However, you're not properly handling these promises. You're just calling `.then(() => {})` which doesn't do anything with the result of the promise or handle any errors that might occur. 

3. Unnecessary use of optional chaining: In line 12, you're using optional chaining (`user?.token`) which is unnecessary because if `userTargetToken` is truthy (checked in line 9), then `user` will always exist in the forEach loop.

Here are the suggested changes:

```diff
const {messaging} = require('firebase-admin');
const FcmTokenFunction = require('../../../databases/functions/fcmToken');
const {FcmToken, User} = require('../../../databases/models');
const UsersFunction = require('../../../databases/functions/users');

const sendChatNotification = async (userTargetId, payload, options = {}) => {
  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(FcmToken, signedUser);
  if (userTargetToken) {
    try {
      const notifications = userTargetToken.map((user) =>
        messaging()
          .sendToDevice(user.token, payload, options)
      );
      await Promise.all(notifications);
    } catch (error) {
      console.error('error send chat notifications', error);
      throw error;
    }
  }
};

module.exports = sendChatNotification;
```

In the updated code, we're using `Promise.all` to wait for all the notifications to be sent before proceeding. This will allow us to catch any errors that occur during the sending of notifications and propagate them up the call stack. We're also throwing the error after logging it, so the caller of the function can handle it appropriately.


<!-- This is an auto-generated comment by OSS CodeRabbit -->